web-api/src/business/useCases/caseDeadline/createCaseDeadlineInteractor.ts

```diff
@@ -12,7 +12,13 @@ import { withLocking } from '@web-api/business/useCaseHelper/acquireLock';
 
 export const createCaseDeadline = async (
   applicationContext: ServerApplicationContext,
-  { caseDeadline }: { caseDeadline: CaseDeadline },
+  {
+    caseDeadline,
+    handlingConsolidatedCases = false,
```

Comment:
I get why this is here (to avoid having to detangle all the stuff happening before the new code ~line 60). That said, I prefer iteration to recursion here. Iteration seems like it would be more readable (for each X, we are doing this thing vs. an if check with conditional recursion based on docketNumbers and leadDocketNumbers), and it would also allow us to more easily create a bunch of deadlines all at once, i.e., one SQL statement rather than each individually.

I don't necessarily think this means the story shouldn't go out; in fact, I think changing this would probably require another light round of regression testing in test. That said, I thought I would at least mention my preference since I think the logic will be difficult for future devs to follow.